lib/services/firebase_service.dart

```diff
@@ -973,35 +987,60 @@ class FirebaseService extends IFirebaseService {
-        recoveryReadiness =
-            formsCollection['formData']['recoveryReadiness'] ?? '-';
-        if (formName == 'Recovery Readiness' &&
-            recoveryReadiness == 'พร้อมฟื้นสภาพหลังผ่าตัด' &&
-            formDateToShow == dateCompare) {
-          evaluationStatus = "completed";
+      if (patientState == "Post-Operation@Hospital") {
+        var state = await checkRecoveryReadinessStatus(hn);
+        if (state == "Pass") {
+          var formStatus = await getFormListInAnBasedOnState(
+              userId: userId, patientState: patientState, formName: formName);
+          if (formStatus.isNotEmpty) {
+            var formsCollection = await _firestore
+                .collection('Forms')
+                .doc(formStatus.last['formId'])
+                .get()
+                .then((value) => value.data())
+                .catchError((onError) {
+              print('$onError no formsCollection');
+            });
+            formCreation = formsCollection['creation'] ?? '-';
+            formTime = DateTime.fromMicrosecondsSinceEpoch(
+                formCreation.microsecondsSinceEpoch);
+            formDateToShow = DateFormat('yyyy-MM-dd').format(formTime);
+            dateToCompare =
+                _calculationService.formatDate(date: DateTime.now());
+            dateCompare = DateFormat('yyyy-MM-dd').format(dateToCompare);
+            if (formDateToShow == dateCompare) {
+              evaluationStatus = "completed";
+            }
+          }
         }
-        if (formName != 'Recovery Readiness' && formDateToShow == dateCompare) {
-          evaluationStatus = "completed";
+        if (state != "Pass") {
+          evaluationStatus = "noPassRecoveryReadiness";
         }
       } else {
-        evaluationStatus = "notCompleted";
+        var formStatus = await getFormListInAnBasedOnState(
+            userId: userId, patientState: patientState, formName: formName);
+        if (formStatus.isNotEmpty) {
+          var formsCollection = await _firestore
+              .collection('Forms')
+              .doc(formStatus.last['formId'])
+              .get()
+              .then((value) => value.data())
+              .catchError((onError) {
+            print('$onError no formsCollection');
+          });
+          formCreation = formsCollection['creation'] ?? '-';
+          formTime = DateTime.fromMicrosecondsSinceEpoch(
+              formCreation.microsecondsSinceEpoch);
+          formDateToShow = DateFormat('yyyy-MM-dd').format(formTime);
+          dateToCompare = _calculationService.formatDate(date: DateTime.now());
+          dateCompare = DateFormat('yyyy-MM-dd').format(dateToCompare);
+
+          if (formDateToShow == dateCompare) {
+            evaluationStatus = "completed";
+          }
+        } else {
+          evaluationStatus = "notCompleted";
+        }
       }
     } else if (vitalSignFormTime != null) {
       var formsCollection = await _firestore
```

Comment:
From line 1046-1062. Why do we need to query from the Forms Collection just to take the formId and then query the sane collection again to get one document. Why we can't get the document from the first query. ?

Reply:
@Thaweesak-sai I already refactor finished, you can check it at 492b8e1